Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -966,9 +959,11 @@ theorem prod_div_prod_filter [CommGroup G] (g : α → M → G) :
 
 end Zero
 
-theorem filter_pos_add_filter_neg [AddZeroClass M] (f : α →₀ M) (p : α → Prop) :
+theorem filter_pos_add_filter_neg [AddZeroClass M] (f : α →₀ M) (p : α → Prop) [DecidablePred p] :
```

Comment:
I think that we should also assume `DecidablePred (¬p ·)` so that this lemma applies in case if the second `filter` uses a classical instance.

Reply:
Otherwise LGTM. Thanks!
bors d+

Reply:
The current statement matches `DFinsupp,filter_pos_add_filter_neg`, so I'd prefer to declare this out of scope. I think we should try doing this to every that uses `¬p` at once, and see if anything breaks: in a separate PR.